conans/client/installer.py

```diff
@@ -488,14 +481,14 @@ def _handle_node_cache(self, node, keep_build, processed_package_references, rem
         output = conanfile.output
 
         layout = self._cache.package_layout(pref.ref, conanfile.short_paths)
-        package_folder = layout.package(pref)
 
         with layout.package_lock(pref):
             if pref not in processed_package_references:
                 processed_package_references.add(pref)
                 if node.binary == BINARY_BUILD:
                     assert node.prev is None, "PREV for %s to be built should be None" % str(pref)
-                    with set_dirty_context_manager(package_folder):
+                    layout.package_remove(pref)
```

Comment:
This is the place of removal of the package folder, instead of inside the ``_build_package()``. This has been changed because the ``set_dirty_context_manager()`` could leave it in an inconsistent state, when removing the package folder, could remove accidentally the .dirty flag.